src/active_item_cache.h

```diff
@@ -9,32 +9,52 @@
 #include "enums.h"
 #include "item.h"
 #include "item_stack.h"
+#include "safe_reference.h"
 
 // A struct used to uniquely identify an item within a submap or vehicle.
 struct item_reference {
     point location;
-    item_stack::iterator item_iterator;
-    // Do not access this from outside this module, it is only used as an ID for active_item_set.
-    item *item_id;
+    safe_reference<item> safe_ref;
```

Comment:
A minor point which would be annoying to change, but I think `item_ref` is a better name.  The fact that it's safe is irrelevant to almost all the code.

Reply:
Since you pointed it out, I had to fix it.